src/main/java/appeng/client/render/tesr/CableBusFastTESR.java

```diff
@@ -0,0 +1,306 @@
+import appeng.api.util.AEAxisAlignedBB;
+import appeng.block.networking.BlockCableBus;
+import appeng.client.render.cablebus.CableBusRenderState;
+import appeng.client.render.cablebus.CubeBuilder;
+import appeng.client.render.cablebus.FacadeBuilder;
+import appeng.client.render.cablebus.FacadeBuilder.TextureAtlasAndTint;
+import appeng.client.render.cablebus.FacadeRenderState;
+import appeng.tile.AEBaseTile;
+
+
+public class CableBusFastTESR extends FastTESR<AEBaseTile>
+{
+	@Override
+	public void renderTileEntityFast( AEBaseTile te, double x, double y, double z, float partialTicks, int destroyStage, float partial, BufferBuilder buffer )
+	{
+		final BlockPos pos = te.getPos();
+		final IBlockAccess world = MinecraftForgeClient.getRegionRenderCache( te.getWorld(), pos );
+
+		CableBusRenderState renderState = getRenderingState( world, pos );
+		if( renderState == null )
+		{
+			return;
+		}
+
+		boolean thinFacades = FacadeBuilder.isUseThinFacades( renderState.getBoundingBoxes() );
+		long rand = MathHelper.getPositionRandom( pos );
+
+		renderState.getFacades().forEach( ( side, state ) ->
+		{
+			AxisAlignedBB facadeBox = FacadeBuilder.getFacadeBox( side, thinFacades );
+			AEAxisAlignedBB cutOutBox = FacadeBuilder.getCutOutBox( facadeBox, renderState.getBoundingBoxes() );
+			List<BakedQuad> quads = this.addFacade(
+					MinecraftForgeClient.getRenderLayer(),
+					renderState.getFacades(),
+					side,
+					cutOutBox,
+					thinFacades,
+					rand );
+
+			buffer.setTranslation( x, y, z );
+			VertexLighterFlat l = new VertexLighterSmoothAo( Minecraft.getMinecraft().getBlockColors() );
+			l.setParent( new VertexBufferConsumer( buffer ) );
+			l.setWorld( world );
+			l.setBlockPos( pos );
+			l.setState( state.getSourceBlock() );
+			l.updateBlockInfo();
+
+			for( BakedQuad q : quads )
+			{
+				q.pipe( l );
+			}
+		} );
+	}
+
+	private static CableBusRenderState getRenderingState( final IBlockAccess world, BlockPos pos )
+	{
+		IBlockState blockState = world.getBlockState( pos );
+		try
+		{
+			blockState = blockState.getActualState( world, pos );
+		}
+		catch( Exception e )
+		{
+			// ignore
+		}
+
+		try
+		{
+			blockState = blockState.getBlock().getExtendedState( blockState, world, pos );
+		}
+		catch( Exception e )
+		{
+			// ignore
+		}
+
+		if( blockState == null || !( blockState instanceof IExtendedBlockState ) )
+		{
+			return null;
+		}
+
+		IExtendedBlockState extendedBlockState = (IExtendedBlockState) blockState;
+		return extendedBlockState.getValue( BlockCableBus.RENDER_STATE_PROPERTY );
+	}
+
+	private List<BakedQuad> addFacade( BlockRenderLayer layer, Map<EnumFacing, FacadeRenderState> facades, EnumFacing side, AEAxisAlignedBB busBounds, boolean thinFacades, long rand )
+	{
+		ArrayList<BakedQuad> ret = new ArrayList<>();
+
+		CubeBuilder builder = new CubeBuilder( DefaultVertexFormats.ITEM, ret );
+		FacadeRenderState facadeState = facades.get( side );
+		IBlockState blockState = facadeState.getSourceBlock();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Move the declaration of "blockState" closer to the code that uses it. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1941)